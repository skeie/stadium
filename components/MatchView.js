// @flow

import React from 'react';
import { Entypo, MaterialCommunityIcons } from '@expo/vector-icons';
import { TouchableOpacity, ScrollView, View, Image, Dimensions } from 'react-native';
import colors from '../constants/Colors';
import { Poppins } from './StyledText';

const { height } = Dimensions.get('window');

export type Match = {
    stadium: { capacity: number, name: string },
    startTime: string,
    homeTeam: string,
    awayTeam: string,
    goalsHomeTeam: number,
    goalsAwayTeam: number,
    goalScorers: Array<{ name: string, minute: Array<string>, team: string }>,
    uri: string,
    date: string,
};

export type Props = Match & {
    onPostMatch: Match => void,
    isEdit: boolean,
};

type GoalScorer = {
    name: string,
    minute: Array<string>,
    team: string,
};

const MatchView = (props: Props) => {
    return (
        <View flex={1} backgroundColor={colors.primary}>
            <ScrollView
                contentContainerStyle={{
                    height: height / 1.5,
                    justifyContent: 'space-around',
                    backgroundColor: colors.primary,
                }}>
                <View
                    flexDirection="column"
                    padding={5}
                    justifyContent="space-around"
                    minHeight="10%">
                    <View flexDirection="row">
                        <Entypo name="location-pin" size={25} color="#553555" />
                        <Poppins style={{ marginLeft: 2 }} type="header">
                            {props.stadium.name}
                        </Poppins>
                    </View>
                    <View marginLeft={27} marginRight={5} flex={1}>
                        <View
                            flexDirection="row"
                            alignItems="center"
                            justifyContent="space-between">
                            <Poppins>Capacity: {props.stadium.capacity}</Poppins>
                            <Poppins>{props.date}</Poppins>
                        </View>
                        <Poppins>Home team: {props.homeTeam}</Poppins>
                    </View>
                </View>
                <Image
                    style={{ width: '100%', height: '50%' }}
                    source={{
                        uri: props.uri,
                    }}
                />
                <View width="100%" flexDirection="row" justifyContent="space-around" padding={10}>
                    <TeamUI team={props.homeTeam} goalScorers={props.goalScorers} />
                    <Poppins type="header">{props.goalsHomeTeam}</Poppins>
                    <Poppins type="header">-</Poppins>
                    <Poppins type="header" style={{ marginRight: 5 }}>
                        {props.goalsAwayTeam}
                    </Poppins>
                    <TeamUI team={props.awayTeam} goalScorers={props.goalScorers} second />
                </View>
            </ScrollView>
        </View>
    );
};

const getGoalScorer = (goalScorers: Array<GoalScorer>, team: string) =>
    goalScorers.filter(goalScorer => goalScorer.team === team);

const TeamUI = ({ team, goalScorers }) => (
    <View flex={1} alignItems="center">
        <Poppins type="header">{team}</Poppins>
        {getGoalScorer(goalScorers, team).map((goalscorer, index) => (
            <View flexDirection="row" alignItems="center" key={index}>
                <MaterialCommunityIcons name="soccer" size={18} color="#553555" />
                <Poppins style={{ marginHorizontal: 3 }}>{goalscorer.name}</Poppins>
                <Poppins>{`(${goalscorer.minute.join(', ')})`}</Poppins>
            </View>
        ))}
    </View>
);

export default MatchView;
