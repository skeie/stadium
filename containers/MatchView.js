// @flow

import React, { Component } from 'react';
import MatchViewUI from '../components/MatchView';
import { post } from '../api/fetch';

class MatchView extends Component<*, *> {
    handlePostMatch = async () => {
        const { isEdit, ...match } = this.props;
        const newGridView = await post('/matches', { ...match });
        this.props.navigation.navigate('Homescreen', { ...newGridView });
    };

    render() {
        return <MatchViewUI onPostMatch={this.handlePostMatch} {...this.props} />;
    }
}

export default MatchView;
