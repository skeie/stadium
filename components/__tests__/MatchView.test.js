import 'react-native';
import React from 'react';
import MatchView from '../MatchView';
import renderer from 'react-test-renderer';
import Top from '../MatchView/Top';
import TeamUI from '../MatchView/TeamUI';
import ScoreView from '../MatchView/ScoreView';

const props = {
    "editable": true,
    "stadium": { "capacity": 121749, "name": "Camp Nou", "__typename": "Stadium" },
    "startTime": "20:45", "homeTeam": "Barcelona", "awayTeam": "Roma", "goalsHomeTeam": 6, "goalsAwayTeam": 1, "goalScorers": [{ "name": "L. SuÃ¡rez", "minute": ["13"], "team": "Barcelona", "__typename": "GoalScorer" }, { "name": "L. Messi", "minute": ["13"], "team": "Barcelona", "__typename": "GoalScorer" }, { "name": "G. Pique", "minute": ["13"], "team": "Barcelona", "__typename": "GoalScorer" }, { "name": "Adriano", "minute": ["13"], "team": "Barcelona", "__typename": "GoalScorer" }, { "name": "E. Dzeko", "minute": ["13"], "team": "Roma", "__typename": "GoalScorer" }], "__typename": "Match", "uri": "https://files.graph.cool/cjdizt45h14ca016541zn4b91/cjdj0ihen0ris01022fxo8dkg", "date": "2017/11/18",
    onChangeHomeTeam: jest.fn(),
}
describe('Matchview', () => {
    it('MatchView renders correctly', () => {
        const tree = renderer.create(<MatchView {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('MatchView top renders correctly with edit true', () => {
        const tree = renderer.create(<Top {...props.stadium} homeTeam={props.homeTeam} date={props.date} onChangeHomeTeam={jest.fn()} editable />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('MatchView top renders correctly with edit false', () => {
        const tree = renderer.create(<Top {...props.stadium} homeTeam={props.homeTeam} date={props.date} onChangeHomeTeam={jest.fn()} editable />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('MatchView TeamUI renders correctly', () => {
        const tree = renderer.create(<TeamUI goalScorers={props.goalScorers} team={props.homeTeam} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('MatchView ScoreView renders correctly', () => {
        const tree = renderer.create(<ScoreView {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
})

