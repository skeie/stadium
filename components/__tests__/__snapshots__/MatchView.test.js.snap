// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matchview MatchView ScoreView renders correctly 1`] = `
<View
  flexDirection="row"
  justifyContent="space-around"
  padding={10}
  width="100%"
>
  <View
    alignItems="center"
    flex={1}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#553555",
            "fontFamily": "poppins",
          },
          Object {
            "color": "#553555",
            "fontFamily": "poppins",
            "fontSize": 20,
          },
          undefined,
        ]
      }
      type="header"
    >
      Barcelona
    </Text>
    <View
      alignItems="center"
      flexDirection="row"
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#553555",
              "fontSize": 18,
            },
            undefined,
            Object {
              "fontFamily": "material-community",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#553555",
              "fontFamily": "poppins",
            },
            undefined,
            Object {
              "marginHorizontal": 3,
            },
          ]
        }
      >
        L. Suárez
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#553555",
              "fontFamily": "poppins",
            },
            undefined,
            undefined,
          ]
        }
      >
        (13)
      </Text>
    </View>
    <View
      alignItems="center"
      flexDirection="row"
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#553555",
              "fontSize": 18,
            },
            undefined,
            Object {
              "fontFamily": "material-community",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#553555",
              "fontFamily": "poppins",
            },
            undefined,
            Object {
              "marginHorizontal": 3,
            },
          ]
        }
      >
        L. Messi
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#553555",
              "fontFamily": "poppins",
            },
            undefined,
            undefined,
          ]
        }
      >
        (13)
      </Text>
    </View>
    <View
      alignItems="center"
      flexDirection="row"
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#553555",
              "fontSize": 18,
            },
            undefined,
            Object {
              "fontFamily": "material-community",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#553555",
              "fontFamily": "poppins",
            },
            undefined,
            Object {
              "marginHorizontal": 3,
            },
          ]
        }
      >
        G. Pique
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#553555",
              "fontFamily": "poppins",
            },
            undefined,
            undefined,
          ]
        }
      >
        (13)
      </Text>
    </View>
    <View
      alignItems="center"
      flexDirection="row"
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#553555",
              "fontSize": 18,
            },
            undefined,
            Object {
              "fontFamily": "material-community",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#553555",
              "fontFamily": "poppins",
            },
            undefined,
            Object {
              "marginHorizontal": 3,
            },
          ]
        }
      >
        Adriano
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#553555",
              "fontFamily": "poppins",
            },
            undefined,
            undefined,
          ]
        }
      >
        (13)
      </Text>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#553555",
          "fontFamily": "poppins",
        },
        Object {
          "color": "#553555",
          "fontFamily": "poppins",
          "fontSize": 20,
        },
        undefined,
      ]
    }
    type="header"
  >
    6
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#553555",
          "fontFamily": "poppins",
        },
        Object {
          "color": "#553555",
          "fontFamily": "poppins",
          "fontSize": 20,
        },
        undefined,
      ]
    }
    type="header"
  >
    -
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#553555",
          "fontFamily": "poppins",
        },
        Object {
          "color": "#553555",
          "fontFamily": "poppins",
          "fontSize": 20,
        },
        Object {
          "marginRight": 5,
        },
      ]
    }
    type="header"
  >
    1
  </Text>
  <View
    alignItems="center"
    flex={1}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#553555",
            "fontFamily": "poppins",
          },
          Object {
            "color": "#553555",
            "fontFamily": "poppins",
            "fontSize": 20,
          },
          undefined,
        ]
      }
      type="header"
    >
      Roma
    </Text>
    <View
      alignItems="center"
      flexDirection="row"
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#553555",
              "fontSize": 18,
            },
            undefined,
            Object {
              "fontFamily": "material-community",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#553555",
              "fontFamily": "poppins",
            },
            undefined,
            Object {
              "marginHorizontal": 3,
            },
          ]
        }
      >
        E. Dzeko
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#553555",
              "fontFamily": "poppins",
            },
            undefined,
            undefined,
          ]
        }
      >
        (13)
      </Text>
    </View>
  </View>
</View>
`;

exports[`Matchview MatchView TeamUI renders correctly 1`] = `
<View
  alignItems="center"
  flex={1}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#553555",
          "fontFamily": "poppins",
        },
        Object {
          "color": "#553555",
          "fontFamily": "poppins",
          "fontSize": 20,
        },
        undefined,
      ]
    }
    type="header"
  >
    Barcelona
  </Text>
  <View
    alignItems="center"
    flexDirection="row"
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#553555",
            "fontSize": 18,
          },
          undefined,
          Object {
            "fontFamily": "material-community",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#553555",
            "fontFamily": "poppins",
          },
          undefined,
          Object {
            "marginHorizontal": 3,
          },
        ]
      }
    >
      L. Suárez
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#553555",
            "fontFamily": "poppins",
          },
          undefined,
          undefined,
        ]
      }
    >
      (13)
    </Text>
  </View>
  <View
    alignItems="center"
    flexDirection="row"
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#553555",
            "fontSize": 18,
          },
          undefined,
          Object {
            "fontFamily": "material-community",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#553555",
            "fontFamily": "poppins",
          },
          undefined,
          Object {
            "marginHorizontal": 3,
          },
        ]
      }
    >
      L. Messi
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#553555",
            "fontFamily": "poppins",
          },
          undefined,
          undefined,
        ]
      }
    >
      (13)
    </Text>
  </View>
  <View
    alignItems="center"
    flexDirection="row"
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#553555",
            "fontSize": 18,
          },
          undefined,
          Object {
            "fontFamily": "material-community",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#553555",
            "fontFamily": "poppins",
          },
          undefined,
          Object {
            "marginHorizontal": 3,
          },
        ]
      }
    >
      G. Pique
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#553555",
            "fontFamily": "poppins",
          },
          undefined,
          undefined,
        ]
      }
    >
      (13)
    </Text>
  </View>
  <View
    alignItems="center"
    flexDirection="row"
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#553555",
            "fontSize": 18,
          },
          undefined,
          Object {
            "fontFamily": "material-community",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#553555",
            "fontFamily": "poppins",
          },
          undefined,
          Object {
            "marginHorizontal": 3,
          },
        ]
      }
    >
      Adriano
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#553555",
            "fontFamily": "poppins",
          },
          undefined,
          undefined,
        ]
      }
    >
      (13)
    </Text>
  </View>
</View>
`;

exports[`Matchview MatchView renders correctly 1`] = `
<View
  backgroundColor="#ADF1D2"
  flex={1}
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "backgroundColor": "#ADF1D2",
        "height": 889.3333333333334,
        "justifyContent": "space-around",
      }
    }
  >
    <View>
      <View
        flexDirection="column"
        justifyContent="space-around"
        minHeight="10%"
        padding={5}
      >
        <View
          flexDirection="row"
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#553555",
                  "fontFamily": "poppins",
                },
                Object {
                  "color": "#553555",
                  "fontFamily": "poppins",
                  "fontSize": 20,
                },
                Object {
                  "marginLeft": 2,
                },
              ]
            }
            type="header"
          >
            Camp Nou
          </Text>
        </View>
        <View
          marginLeft={27}
          marginRight={5}
        >
          <View
            alignItems="center"
            flexDirection="row"
            justifyContent="space-between"
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#553555",
                    "fontFamily": "poppins",
                  },
                  undefined,
                  undefined,
                ]
              }
            >
              Capacity: 
              121749
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#553555",
                    "fontFamily": "poppins",
                  },
                  undefined,
                  undefined,
                ]
              }
            >
              2017/11/18
            </Text>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginLeft": 5,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          />
          <View
            marginRight={8}
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#553555",
                  "fontFamily": "poppins",
                },
                undefined,
                undefined,
              ]
            }
          >
            Home team: 
            Barcelona
          </Text>
        </View>
      </View>
      <Image
        source={
          Object {
            "uri": "https://files.graph.cool/cjdizt45h14ca016541zn4b91/cjdj0ihen0ris01022fxo8dkg",
          }
        }
        style={
          Object {
            "height": "50%",
            "width": "100%",
          }
        }
      />
      <View
        flexDirection="row"
        justifyContent="space-around"
        padding={10}
        width="100%"
      >
        <View
          alignItems="center"
          flex={1}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#553555",
                  "fontFamily": "poppins",
                },
                Object {
                  "color": "#553555",
                  "fontFamily": "poppins",
                  "fontSize": 20,
                },
                undefined,
              ]
            }
            type="header"
          >
            Barcelona
          </Text>
          <View
            alignItems="center"
            flexDirection="row"
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#553555",
                    "fontFamily": "poppins",
                  },
                  undefined,
                  Object {
                    "marginHorizontal": 3,
                  },
                ]
              }
            >
              L. Suárez
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#553555",
                    "fontFamily": "poppins",
                  },
                  undefined,
                  undefined,
                ]
              }
            >
              (13)
            </Text>
          </View>
          <View
            alignItems="center"
            flexDirection="row"
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#553555",
                    "fontFamily": "poppins",
                  },
                  undefined,
                  Object {
                    "marginHorizontal": 3,
                  },
                ]
              }
            >
              L. Messi
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#553555",
                    "fontFamily": "poppins",
                  },
                  undefined,
                  undefined,
                ]
              }
            >
              (13)
            </Text>
          </View>
          <View
            alignItems="center"
            flexDirection="row"
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#553555",
                    "fontFamily": "poppins",
                  },
                  undefined,
                  Object {
                    "marginHorizontal": 3,
                  },
                ]
              }
            >
              G. Pique
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#553555",
                    "fontFamily": "poppins",
                  },
                  undefined,
                  undefined,
                ]
              }
            >
              (13)
            </Text>
          </View>
          <View
            alignItems="center"
            flexDirection="row"
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#553555",
                    "fontFamily": "poppins",
                  },
                  undefined,
                  Object {
                    "marginHorizontal": 3,
                  },
                ]
              }
            >
              Adriano
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#553555",
                    "fontFamily": "poppins",
                  },
                  undefined,
                  undefined,
                ]
              }
            >
              (13)
            </Text>
          </View>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#553555",
                "fontFamily": "poppins",
              },
              Object {
                "color": "#553555",
                "fontFamily": "poppins",
                "fontSize": 20,
              },
              undefined,
            ]
          }
          type="header"
        >
          6
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#553555",
                "fontFamily": "poppins",
              },
              Object {
                "color": "#553555",
                "fontFamily": "poppins",
                "fontSize": 20,
              },
              undefined,
            ]
          }
          type="header"
        >
          -
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#553555",
                "fontFamily": "poppins",
              },
              Object {
                "color": "#553555",
                "fontFamily": "poppins",
                "fontSize": 20,
              },
              Object {
                "marginRight": 5,
              },
            ]
          }
          type="header"
        >
          1
        </Text>
        <View
          alignItems="center"
          flex={1}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#553555",
                  "fontFamily": "poppins",
                },
                Object {
                  "color": "#553555",
                  "fontFamily": "poppins",
                  "fontSize": 20,
                },
                undefined,
              ]
            }
            type="header"
          >
            Roma
          </Text>
          <View
            alignItems="center"
            flexDirection="row"
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#553555",
                    "fontFamily": "poppins",
                  },
                  undefined,
                  Object {
                    "marginHorizontal": 3,
                  },
                ]
              }
            >
              E. Dzeko
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#553555",
                    "fontFamily": "poppins",
                  },
                  undefined,
                  undefined,
                ]
              }
            >
              (13)
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`Matchview MatchView top renders correctly with edit false 1`] = `
<View
  flexDirection="column"
  justifyContent="space-around"
  minHeight="10%"
  padding={5}
>
  <View
    flexDirection="row"
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#553555",
            "fontSize": 25,
          },
          undefined,
          Object {
            "fontFamily": "entypo",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#553555",
            "fontFamily": "poppins",
          },
          Object {
            "color": "#553555",
            "fontFamily": "poppins",
            "fontSize": 20,
          },
          Object {
            "marginLeft": 2,
          },
        ]
      }
      type="header"
    >
      Camp Nou
    </Text>
  </View>
  <View
    marginLeft={27}
    marginRight={5}
  >
    <View
      alignItems="center"
      flexDirection="row"
      justifyContent="space-between"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#553555",
              "fontFamily": "poppins",
            },
            undefined,
            undefined,
          ]
        }
      >
        Capacity: 
        121749
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#553555",
              "fontFamily": "poppins",
            },
            undefined,
            undefined,
          ]
        }
      >
        2017/11/18
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginLeft": 5,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 15,
          },
          undefined,
          Object {
            "fontFamily": "material",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <View
      marginRight={8}
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#553555",
            "fontFamily": "poppins",
          },
          undefined,
          undefined,
        ]
      }
    >
      Home team: 
      Barcelona
    </Text>
  </View>
</View>
`;

exports[`Matchview MatchView top renders correctly with edit true 1`] = `
<View
  flexDirection="column"
  justifyContent="space-around"
  minHeight="10%"
  padding={5}
>
  <View
    flexDirection="row"
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#553555",
            "fontSize": 25,
          },
          undefined,
          Object {
            "fontFamily": "entypo",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#553555",
            "fontFamily": "poppins",
          },
          Object {
            "color": "#553555",
            "fontFamily": "poppins",
            "fontSize": 20,
          },
          Object {
            "marginLeft": 2,
          },
        ]
      }
      type="header"
    >
      Camp Nou
    </Text>
  </View>
  <View
    marginLeft={27}
    marginRight={5}
  >
    <View
      alignItems="center"
      flexDirection="row"
      justifyContent="space-between"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#553555",
              "fontFamily": "poppins",
            },
            undefined,
            undefined,
          ]
        }
      >
        Capacity: 
        121749
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#553555",
              "fontFamily": "poppins",
            },
            undefined,
            undefined,
          ]
        }
      >
        2017/11/18
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginLeft": 5,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 15,
          },
          undefined,
          Object {
            "fontFamily": "material",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <View
      marginRight={8}
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#553555",
            "fontFamily": "poppins",
          },
          undefined,
          undefined,
        ]
      }
    >
      Home team: 
      Barcelona
    </Text>
  </View>
</View>
`;
